Change orm, fixed dockerfile and etc

lunasy26 opened the pull request:
1. Changed orm from sql server to postresql
2. Added an extension method
3. Fixed dockerfile

Commit:
1. Changed orm from sql server to postresql
2. Added an extension method
3. Fixed dockerfile

Kharlap-Sergey commented on src/Infrastructure/Memo4me.Persistence/Extensions.cs:
```diff
@@ -0,0 +1,15 @@
+    {
+        services.AddDbContext<AppDbContext>(x =>
+        {
+            x.UseNpgsql("Username=postgres;Password=123;Host=db;Port=5432;Database=MemoDb");
```
should be passed as a configuration parameter

Kharlap-Sergey replied:
Read more about configuration https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-8.0

Kharlap-Sergey commented on devops/Dockerfile:
```diff
@@ -0,0 +1,33 @@
+# Копируем собранное приложение из предыдущего шага
+COPY --from=build /app/out ./
+
+# Открываем порт 80 для веб-приложения
```
redundant code

Kharlap-Sergey commented on devops/Dockerfile:
```diff
@@ -0,0 +1,33 @@
+﻿# Используем официальный образ .NET SDK для сборки приложения
```
Russian comments 🥲

Kharlap-Sergey commented on devops/Dockerfile:
```diff
@@ -0,0 +1,33 @@
+COPY src/Core/Memo4me.Domain/*.csproj ./src/Core/Memo4me.Domain/
+COPY src/Infrastructure/Memo4me.Persistence/*.csproj ./src/Infrastructure/Memo4me.Persistence/
+COPY src/Presentation/Memo4me.API/*.csproj ./src/Presentation/Memo4me.API/
+COPY src/Presentation/Memo4me.UI/*.csproj ./src/Presentation/Memo4me.UI/
```
I don't see any usages for this UI projects, and I believe you don't have references on it

Kharlap-Sergey commented on devops/Dockerfile:
BTW, from where do I need to run docker build command to get image?

Commit:
fix: resolve pull request review comments

- Moved connection string configuration to environment variables
- Ensured Swagger displays correctly after docker build in development mode

Commit:
docs: add Docker build and run steps to README

Kharlap-Sergey commented on README.md:
```diff
@@ -1 +1,14 @@
+
+### Building and Running the Docker Containers
+
+1. Navigate to the `App` directory
```
There is no app directory in the repo.
It is common practice to give advice assuming that a user is  already in a repo and call it root

Kharlap-Sergey commented on README.md:
```diff
@@ -1 +1,14 @@
+
+2. Build the Docker image:
+   
+    docker build -t memo4me-image -f devops/Dockerfile .
```
commands can be wrapped with `` to emphasise on them like: `dotnet restore`

Kharlap-Sergey commented on devops/Dockerfile:
```diff
@@ -1,33 +1,31 @@
 COPY src/Infrastructure/Memo4me.Persistence/*.csproj ./src/Infrastructure/Memo4me.Persistence/
 COPY src/Presentation/Memo4me.API/*.csproj ./src/Presentation/Memo4me.API/
-COPY src/Presentation/Memo4me.UI/*.csproj ./src/Presentation/Memo4me.UI/
+#COPY src/Presentation/Memo4me.UI/*.csproj ./src/Presentation/Memo4me.UI/
```
Can we remove it, as there is no reason to keep it commented?

Kharlap-Sergey commented on src/Infrastructure/Memo4me.Persistence/Extensions.cs:
```diff
@@ -1,15 +1,16 @@
         services.AddDbContext<AppDbContext>(x =>
         {
-            x.UseNpgsql("Username=postgres;Password=123;Host=db;Port=5432;Database=MemoDb");
+            x.UseNpgsql(configuration.GetConnectionString("PostgresConnection"));
```
I would've provided a connection string directly to method.

Commit:
fix: complete adjustments based on pull request feedback

- Updated README instructions to assume user starts in repo root
- Refactored AddPersistence method to inject connection string directly as parameter
